sdk/core/src/CommonNetworkMember.ts

```diff
@@ -95,30 +96,50 @@ import {
   PROD_METRICS_URL,
 } from './_defaultConfig'
 
+type Constructor<T> = new (
+  password: string,
+  encryptedSeed: string,
+  options?: SdkOptions,
+  component?: EventComponent,
+) => T
+type AbstractStaticMethods<T> = {
+  afterConfirmSignUp: (networkMember: T, originalOptions: SdkOptions) => Promise<void>
+}
+type ConstructorKeys<T> = {
+  [P in keyof T]: T[P] extends new (...args: unknown[]) => unknown ? P : never
+}[keyof T]
+type OmitConstructor<T> = Omit<T, ConstructorKeys<T>>
+type DerivedThis<T> = Constructor<T> & AbstractStaticMethods<T> & OmitConstructor<typeof CommonNetworkMember>
+
 @profile()
-export class CommonNetworkMember {
+export abstract class CommonNetworkMember {
```

Comment:
@inga-affinidi lets not do this class as abstract, at this point, cause I believe right now its could be used at different places at customer on backends as separate instance.
Lets mention at the CHANGE_LOG (And Probably at REAMDE as well) that its going to happen as abstract in the near future, and will be not available. And that consumers should use browser SDK where there using core SDK instead, and make proper announcement at the network_release channel about that (please post that this attention message there, when this version will be released)

Reply:
It has to be abstract for a reason, in order for static factories to work properly.

After this PR, `CommonNetworkMember` constructor will have `platformNetworkTools` as one of its parameters; but for the sake of simplicity, all the factory methods (e.g. `CommonNetworkMember.signUp`) do not accept `platformNetworkTools` as a parameter, instead requiring `this` constructor to be
```
new (
  password: string,
  encryptedSeed: string,
  options?: SdkOptions,
  component?: EventComponent,
) => T
```

So we could have `AffinidiWallet extends CommonNetworkMember` with its own constructor with that simpler signature (which will pass its platform tools to `CommonNetworkMember` constructor); and `AffinidiWallet.signUp` will automatically return an instance of `AffinidiWallet` with all the types declared properly, even though `AffinidiWallet` does not have a static `signUp` member.

On the other hand, it means that direct calls `CommonNetworkMember.signUp` won't work (and TypeScript won't let to make them), because `CommonNetworkMember` does not have a constructor of that simpler required type.

(We're doing some java-style OOP here, but it is worth it because it allows to remove all that complexity; a lot of methods had 3-4 identical copies before that PR in order to make everything work, and still there were some problems with types).

---

Why do customers want to use `CommonNetworkMember` directly?

If they don't need features that depend on platform encryption tools, they could always do something `common` now does: https://github.com/affinityproject/affinidi-core-sdk/blob/985b097422f40d2f9ba4a4acb2a2ca9e3be17597/common-libs/common/test/factory/CommonNetworkMember.ts

If they need these features, they will have to provide platform encryption tools anyway; and if e.g. platform encryption tools from `wallet-browser-sdk` work on their platform, why won't they just use `AffinidiWallet` from `wallet-browser-sdk`?

I understand that this is a breaking change, which is why I've bumped the version all the way up to 5.0.0. But I agree that we might want to clearly describe the migration path, if you have some specific usecases in mind.

Reply:
Yeah, to have proper instance of classe as well, to be able get rid of this hack.
Ok, to avoid extra work for this soft version bump - then lets hihligh this at the channel and at the Chagelog explicitly with proper way what should be updated if somebody using core-sdk as it is right now